lib/Mongo/MongoClient.php

```diff
@@ -229,7 +262,10 @@ public function listDBs()
      */
     public function selectCollection($db, $collection)
     {
-        return new MongoCollection($this->selectDB($db), $collection);
+        $collection = new MongoCollection($this->selectDB($db), $collection);
+        $collection->setReadPreference($this->readPreference['type'], $this->readPreference['tagsets']);
```

Comment:
What I meant was that instead of `selectCollection` and `selectDB` applying `readPreference` and `writeConcern` to the collection and database classes the constructors for `MongoCollection` and `MongoDB` should just grab them from the given `MongoDB` and `MongoClient` objects, respectively. This way `readPreference` and `writeConcern` would also be applied when doing this:
```
$client = new MongoClient();
$client->setReadPreference('secondaryPreferred');
$db = new MongoDB($client, 'mongo-php-adapter');
```